crates/bevy_asset/src/asset_server.rs

```diff
@@ -558,6 +558,41 @@ impl AssetServer {
             }
         }
     }
+    /// Loads an asset and blocks until it either loads of fails to load. Useful for loading
+    /// default assets and for when startup times are a non issue such as when prototyping.
+    pub fn block_on(
+        &self,
+        handle: impl Into<HandleUntyped>,
+    ) -> Result<HandleUntyped, AssetServerError> {
+        let mut handle = handle.into();
+        let path =
+            self.get_handle_path(handle.clone())
+                .ok_or(AssetServerError::AssetLoaderError(anyhow::anyhow!(
+                    "Handle does not have a corresponding Path"
+                )))?;
+        'block_on: loop {
+            match self.get_load_state(handle.clone()) {
+                LoadState::NotLoaded => {
+                    handle = self.load_untyped(path.clone());
+                    continue 'block_on;
+                }
+                LoadState::Loading => {
+                    continue 'block_on;
+                }
```

Comment:
As this is a busy loop, i would suggest adding a [yield_now](https://doc.rust-lang.org/std/thread/fn.yield_now.html) or [spin_loop](https://doc.rust-lang.org/std/hint/fn.spin_loop.html) call here. (Not sure which is better here)

Reply:
From reading the docs of both of those a yield_now seems more appropriate because of the comment in the spin_loop  docs that "it is strongly recommended that the spin loop is terminated after a finite amount of iterations"